src/model/reduction_modes.h

```diff
@@ -53,5 +53,57 @@ void reduceModeForce(
 
 }
 
+/*
+ * @bief: reduceModeForce is essentially a dot product of the force vector and the modevector resulting in the effective force from the modes.
+ * IMPORTANT: 1. the forces used have to be rotated such that they are the corresponding coordinatesystem.
+ * 2. after the forces have been reduced they have to corrected by correctModeforce
+ *
+ */
+template<typename REAL>
+void reduceModeForce(
+		const REAL* forceX,
+		const REAL* forceY,
+		const REAL* forceZ,
+		const REAL* modeX,
+		const REAL* modeY,
+		const REAL* modeZ,
+		unsigned const& numAtoms,
+		unsigned const& numModes,
+		REAL* result
+		)
+{
+	//TODO: think about passing protein to function with member "isreceptor"to determine rotation
+	//rotate forces into ligand frame
+	for( int i=0; i<numModes;i++){result[i]=0;}
+	for (unsigned i = 0; i < numAtoms; ++i) {
+		for(int mode=0;mode<numModes;mode++){
+				result[mode] -= forceX[i]*modeX[i*numModes+mode]+forceY[i]*modeY[i*numModes+mode]+forceZ[i]*modeZ[i*numModes+mode];
+		}
+	}
+}
+/**
+ * @brief: this function corrects for strong mode interaction.
+ * In case that high forces are acting the Proteins the mode tend to intoduce too much deformation.
+ * Adding an exponential term which is of higher order (harmonic not enough) which is multiplied by the forceconstant for each mode corrects for this.
+ * The old version uses either exp=3 or exp=4.
+ * The forceconstant correspoonds to the magnitude of the modevector for each mode.
+ */
+template<typename REAL>
+void correctModeForce(
+		const REAL* modeForceConstant,
+		unsigned const& numModes,
+		REAL* delta
+		)
+{
+	const REAL factor=4.0;
```

Comment:
int c++11 make it a constexpr.: constexpr REAL factor=4.0; We later can put these constants into a configuration header

please insert spaces before and after operators in general (here: "y = x"  )